astropy/stats/src/fast_sigma_clip.c

```diff
@@ -99,39 +102,54 @@ static void _sigma_clip_fast(
     npy_intp is_array = *steps++;
     npy_intp is_mask = *steps++;
 
-    double *buffer = NULL;
+    double *data_buffer = NULL;
+    double *mad_buffer = NULL;
 
-    buffer = (double *)PyArray_malloc(n_i * sizeof(double));
-    if (buffer == NULL) {
+    // data_buffer is used to store the current values being sigma clipped
+    data_buffer = (double *)PyArray_malloc(n_i * sizeof(double));
+    if (data_buffer == NULL) {
         PyErr_NoMemory();
         return;
     }
+
+    // mad_buffer is available to be used by compute_sigma_clipped_bounds
```

Comment:
So, I'd move this inside the loop, as follows:
```
if (((npy_bool *)use_mad_std) && mad_buffer == NULL) {
...
}
```

Reply:
Done